front-end-tools/src/Main.tsx

```diff
@@ -478,29 +515,55 @@ export default function Main(props: ConnectionProps) {
                                             type="checkbox"
                                             id="useModuleReferenceFromStep1"
                                             onChange={() => {
+                                                setModuleReferenceError('');
+                                                setModuleReference('');
+                                                setInitName('');
                                                 const checkboxElement = document.getElementById(
                                                     'useModuleReferenceFromStep1'
                                                 ) as HTMLInputElement;
 
+                                                setCheckedBoxElemenChecked(checkboxElement.checked);
+
+                                                const element = document.getElementById(
```

Comment:
This doesn't look right. I'm not seeing any elements with that `id` in the project?

But it also isn't the way you're supposed to do it in React. Either use `ref` or store the contents in component state.